### ISSUE TITLE
Game specific settings

### ISSUE THREAD

**Issue by voidcat0** [Ryujinx/Ryujinx#1796]
#### Overview:
Add separate settings for each game to make it possible to use different configuration settings for each.

#### Smaller Details: 
Keep the main settings as a default setting for new and unconfigured games. Then add the ability to change PPTC, resolution scale, anisotropy, shader cache, [...] and all future settings separately for each game.

#### Nature of Request:  
- Addition
  - Right-click menu for each game entry allowing to configure game specific settings, which are initially taken from the Ryujinx default settings (from the main menu), to tailor them to a specific game.
  - The option to revert the game settings to "default" settings (those defined in Ryujinx, "forgetting" the game specific settings)

### Why would this feature be useful?
Adding this feature would allow some games to be treated with the workaround they need, while not having to change the emulation settings globally for all games.
One example: 
[Hyrule Warriors: Definitive Edition](https://github.com/Ryujinx/Ryujinx-Games-List/issues/168) still crashes/freezes ingame when the PPTC is enabled, but plays fine with it disabled. So every time I want to start that game, I have to disable the PPTC, then enable it after playing for other games to benefit from the PPTC.

**Comment by Silent-Remux** [comment]
Another use case is allowing ultrawide mods while not having to change the global settings from "stretch to window" back to "16:9" for games that don't have ultrawide mods.

**Comment by GamerzHell9137** [comment]
> 

You can easily click here to change the AR on the fly 
![image](https://user-images.githubusercontent.com/2002038/190710834-afcec1dd-790d-44b3-bf07-c458561f1f4e.png)

**Comment by Silent-Remux** [comment]
> > 
> 
> You can easily click here to change the AR on the fly ![image](https://user-images.githubusercontent.com/2002038/190710834-afcec1dd-790d-44b3-bf07-c458561f1f4e.png)

Sure but obviously individual game settings ala Dolphin would be much preferred.

**Comment by Maverynthia** [comment]
Still waiting for this feature. Especially since the new Pokemon games need Vulkan over OpenGL and it's a pain to switch them all the time.

**Comment by saldabain** [comment]
I'm working on this, though I'm really busy nowadays so please be patient.

**Comment by Omegaxis** [comment]
I just relised on my Steam Deck when playing Sonic Frontiers that Open GL works better then Vulcan when it comes to the textures, i don't get as many glitches, and now when am looking for Game specific settings for just Sonic Frontiers only to relies this feature is not in Yet?, how long as this Emulator been out and its 2023, the Heck?

**Comment by MutantAura** [comment]
> I just relised on my Steam Deck when playing Sonic Frontiers that Open GL works better then Vulcan when it comes to the textures, i don't get as many glitches, and now when am looking for Game specific settings for just Sonic Frontiers only to relies this feature is not in Yet?, how long as this Emulator been out and its 2023, the Heck?

The code is open source. If you'd like to implement a feature you'd find useful, then we'd be happy to review and accept it.

**Comment by lolekuk** [comment]
Yes please, would be useful as I have to run some games with/without vsync. Thanks.

**Comment by lBleyl** [comment]
yes, please, this feature is a must have

**Comment by Avrution** [comment]
Still hopeful for this one.  Very useful for graphical configurations between games as well as control mapping.

**Comment by Toushirou90** [comment]
any information about the progress of this project?

- devolepment in progress but be patient?
- project on ice, no active development?
- almost finished, feature will be added in the near future?

**Comment by HoriZonUK** [comment]
This so needed  for example World of Goo wont start in Docked mode as it will crash ryujinx ,  but it loads and runs in Handheld mode fine.

**Comment by OmniiOMEGA** [comment]
> Still waiting for this feature. Especially since the new Pokemon games need Vulkan over OpenGL and it's a pain to switch them all the time.

I’m experiencing this now playing shining pearl but oddly enough getting better performance on OpenGL on latest Ryujinx?

**Comment by OmniiOMEGA** [comment]
@saldabain is this still WIP?

**Comment by pyrates999** [comment]
Just adding that I too would appreciate this being added.

**Comment by Kimchidude** [comment]
More requesting for per game settings.

**Comment by gfrhj** [comment]
This is really important for control options imo I have some really weird input configs for some games and not having to swap it around would just be convenient, especially as I'm on a steam deck. either that or at least make the control config more intuitive. i still don't instant why i can't save over the default when I'm literally never going to use it

**Comment by Optekah** [comment]
One scenario I haven't seen mentioned where this would be hugely beneficial is couch gaming on a TV.

I run all my emulators through ES-DE and so everything I do can only be controlled from a controller. Pain in the ass when something needs changing, so I get up, climb two flights, turn my monitor on, swap to OpenGL, turn the monitor off, go back downstairs, sit down, and then realize that I forgot to full screen it or change the AR or whatever :P

A very much needed feature. Thanks for your hard work, devs!

**Comment by MrSythos** [comment]
For me this would be a nice time saver. It's not like I can't change the settings on a game by game basis when I boot it up, but it would be pretty cool to have it save everything I changed so the next time I load in it will be how I want it. Yuzu was able to do that pretty well, but since that's gone Ryujinx is the only decent Switch emulator still being updated. Hope this gets added at some point.
